### PR TITLE
SPDX license identifier

### PR DESCRIPTION
**Pull request by hudokkow**
# **This is WIP!**

Replaces https://github.com/xbmc/xbmc/pull/13927

Cleanup the mess with copyright notices/replace with an SPDX license identifier. It allows better overview of licenses, for both humans and compliance software.

### Current licenses (as seen on [SPDX licenses](https://spdx.org/licenses/) list):

| License                   | FSF Free/Libre     | OSI Approved       | License link                                                 | Ready              |
| ------------------------- | :----------------: | :----------------: | :----------------------------------------------------------: | :----------------: |
| BSD-3-Clause              | :heavy_check_mark: | :heavy_check_mark: | **[link](https://spdx.org/licenses/BSD-3-Clause.html)**      | :heavy_check_mark: |
| BSD-4-Clause              | :heavy_check_mark: | :heavy_minus_sign: | **[link](https://spdx.org/licenses/BSD-4-Clause.html)**      | :heavy_check_mark: |
| GPL-2.0-only              | :heavy_check_mark: | :heavy_check_mark: | **[link](https://spdx.org/licenses/GPL-2.0-only.html)**      | :heavy_check_mark: |
| GPL-2.0-or-later          | :heavy_check_mark: | :heavy_check_mark: | **[link](https://spdx.org/licenses/GPL-2.0-or-later.html)**  | :heavy_check_mark: |
| LGPL-2.0-or-later         | :heavy_minus_sign: | :heavy_check_mark: | **[link](https://spdx.org/licenses/LGPL-2.0-or-later.html)** | :heavy_check_mark: |
| LGPL-2.1-or-later         | :heavy_check_mark: | :heavy_check_mark: | **[link](https://spdx.org/licenses/LGPL-2.1-or-later.html)** | :heavy_check_mark: |
| MIT                       | :heavy_check_mark: | :heavy_check_mark: | **[link](https://spdx.org/licenses/MIT.html)**               | :heavy_check_mark: |
| PHP-3.01                  | :heavy_check_mark: | :heavy_minus_sign: | **[link](https://spdx.org/licenses/PHP-3.01.html)**          | :heavy_check_mark: |
| Unlicense (Public Domain) | :heavy_check_mark: | :heavy_minus_sign: | **[link](https://spdx.org/licenses/Unlicense.html)**         | :heavy_check_mark: |

[txt licenses](https://github.com/spdx/license-list-data/tree/master/text)

### Method:
1. Read about SPDX license format.
2. Read some more.
3. Follow Linux kernel example (see resources below).
4. Run [scancode](https://github.com/nexB/scancode-toolkit) on master branch `./scancode -l -n 16 ../kodi --html ../kodi_licenses.html`.
5. Identify all licenses spewed by [scancode](https://github.com/nexB/scancode-toolkit).
6. Group files with same license (ignored biggest group for now, **GPL-2.0-or-later**).
7. Confirm group by group, file by file if detected license is correct. This is very important. Many are false positives because an expression or phrase triggers them. That's the case for **all *gpl-1.0-plus***, triggered by ` *  http://www.gnu.org/copyleft/gpl.html`.
8. If license is not clear (missing text, ambiguous interpretation, etc.) try to confirm with original/other sources. That was the case with `xbmc/platform/win32/dxerr.h` and `xbmc/platform/win32/dxerr.cpp` that looked like **MS-PL** licensed files but were indeed **MIT** licensed.
8. After initial assessment, start with the smallest license group and change notice manually.
9. Loop.
10. Finally, take care of the biggest group, **GPL-2.0-or-later**, which is ours. Patterns are easy to find and `sed` is a very good friend.
11. Select a random sample of files and have a look to see if everything is OK.

### Resources used:

* https://spdx.org/
* https://spdx.org/licenses/
* https://spdx.org/tools
* https://github.com/david-a-wheeler/spdx-tutorial
* https://github.com/torvalds/linux/tree/master/LICENSES
* https://www.kernel.org/doc/html/latest/process/license-rules.html
* https://lwn.net/Articles/739183/
* https://lwn.net/Articles/738809/
* https://git.fsfe.org/reuse/reuse
* https://fsfe.org/news/2017/news-20171219-01.en.html
* https://reuse.software/dev/
* https://reuse.software/practices/2.0/
* https://git.fsfe.org/reuse/spdx-hello
* https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses

### CODE REVIEW

**Comment by garbear** [comment]
awesome! should we move the LGPL xbmc code over to GPL for consistency?

**Comment by hudokkow** [comment]
Don't know. I'm not a FOSS license expert, far from it.
Right now I'm only substituting licenses with their SPDX counterpart and trying my best to not fuck up.

Anyway, it's far from finished.

I'll PR the first 2 parts, headers cleanup and nuking trailing space. It's a PITA to rebase this every time a PR goes in.

**Comment by garbear** [comment]
get it in soon. the PR I opened today adds tons of code.

**Comment by hudokkow** [comment]
yeah, saw it just now. Don't worry, I already rebased this 15 times today... :smile:

File: xbmc/contrib/kissfft/_kiss_fft_guts.h [hunk 1]
```diff
@@ -1,16 +1,10 @@
...
-    * Neither the author nor the names of any contributors may be used to endorse or promote products derived from this software without specific prior written permission.
-
-THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
-*/
+ *  Copyright (c) 2003-2010, Mark Borgerding
+ *  All rights reserved.
+ *
+ *  SPDX-License-Identifier: BSD-3-Clause
+ *  See LICENSES/README.md for more information.
+ */
```
Comment (hudokkow) [inline xbmc/contrib/kissfft/_kiss_fft_guts.h]: @mborgerding, sorry to bother you. I need to know if you're OK with ^^.

It's part of the effort to convert Kodi to the **[SPDX license format](https://spdx.org/licenses/)**. Complete set of changes can be seen **[here](https://github.com/hudokkow/xbmc/pull/4)**, if you want some background.

Assuming you're OK with it, would you accept a pull request to convert `kissfft` to the SPDX license format? I already have **[some work done](https://github.com/hudokkow/kissfft/commit/ada7bfae3399e42ca2c89c3c42187425063007f0)**.

Cheers!

Comment (mborgerding) [reply xbmc/contrib/kissfft/_kiss_fft_guts.h]: I'm just catching up to speed on the SPDX effort.  I hadn't heard of it before now.  I think that is a great idea: a step toward solving licensing fatigue. I'd be happy to see your pull request.

Comment (hudokkow) [reply xbmc/contrib/kissfft/_kiss_fft_guts.h]: Great! :+1: I'll open a PR in the next few days.

**Pull request merged**